pkg/query-service/querycache/query_range_cache_test.go

```diff
@@ -315,13 +322,38 @@ func TestMergeWithCachedSeriesData(t *testing.T) {
 	// Call MergeWithCachedSeriesData
 	mergedData := q.MergeWithCachedSeriesData(cacheKey, newData)
 
+	// Sort the series before comparison to avoid flaky test
+	for _, seriesData := range mergedData {
+		sortSeries(seriesData.Data)
+		for _, series := range seriesData.Data {
+			sort.Slice(series.Points, func(i, j int) bool {
+				return series.Points[i].Timestamp < series.Points[j].Timestamp
+			})
+		}
+	}
+
+	for _, seriesData := range expectedMergedData {
+		sortSeries(seriesData.Data)
+		for _, series := range seriesData.Data {
+			sort.Slice(series.Points, func(i, j int) bool {
+				return series.Points[i].Timestamp < series.Points[j].Timestamp
+			})
+		}
+	}
+
+	// Sort expectedMergedData for consistency
+	sort.Slice(expectedMergedData, func(i, j int) bool {
+		return expectedMergedData[i].Start < expectedMergedData[j].Start
```

Comment:
Did you arrive at a case where this was actually failing without sorting ?

Reply:
yes, we can see fails by running this test multiple times
like `go test ./pkg/query-service/querycache -run MergeWithCachedSeriesData -v -count=100`
@nityanandagohain

Reply:
cc @srikanthccv

Reply:
We shouldn't be fixing this in the test; `mergeSeries` from `queryCache` is the reason for flakiness because it uses map. We should solve the problem in `mergeSeries`. 

```diff
+       hashes := make([]uint64, 0, len(seriesesByLabels))
+       for h := range seriesesByLabels {
+               hashes = append(hashes, h)
+       }
+       sort.Slice(hashes, func(i, j int) bool {
+               return hashes[i] < hashes[j]
+       })
+
        // Sort the points in each series by timestamp
-       for idx := range seriesesByLabels {
-               series := seriesesByLabels[idx]
+       for _, h := range hashes {
+               series := seriesesByLabels[h]
                series.SortPoints()
                series.RemoveDuplicatePoints()
                mergedSeries = append(mergedSeries, series)
```

Reply:
yeah you're right...!
I applied the suggested change by 7620088